packages/s3/adapter.ts

```diff
@@ -0,0 +1,72 @@
+import { StorageAdapter } from 'https://lib.deno.dev/x/grammy@1.x/mod.ts';
+import { S3Client } from 'jsr:@bradenmacdonald/s3-lite-client@0.7.4';
+export { S3Client } from 'jsr:@bradenmacdonald/s3-lite-client@0.7.4';
+/**
+ * The type of the constructor argument for S3Client
+ */
+// vendored from https://github.com/bradenmacdonald/s3-lite-client/blob/b34c604d0e9c3741919c92bc0151ec7d13eae467/client.ts#L15C1-L29C2
+// can be removed when https://github.com/bradenmacdonald/s3-lite-client/pull/38 is merged
+export interface S3ClientOptions {
+  /** Hostname of the endpoint. Not a URL, just the hostname with no protocol or port. */
+  endPoint: string;
+  accessKey?: string;
+  secretKey?: string;
+  sessionToken?: string;
+  useSSL?: boolean | undefined;
+  port?: number | undefined;
+  /** Default bucket name, if not specified on individual requests */
+  bucket?: string;
+  /** Region to use, e.g. "us-east-1" */
+  region: string;
+  /** Use path-style requests, e.g. https://endpoint/bucket/object-key instead of https://bucket/object-key (default: true) */
+  pathStyle?: boolean | undefined;
+}
+
+export type S3StorageClient = Pick<
+S3Client,
+'exists' | 'deleteObject' | 'getObject' | 'host' | 'region' | 'putObject'
+>;
+const isS3StorageClient = (
+  maybeClient: S3StorageClient | any,
+): maybeClient is S3StorageClient =>
+  ['exists', 'deleteObject', 'getObject', 'putObject'].every((required) =>
+    typeof maybeClient[required] === 'function'
+  );
```

Comment:
Please use code block.

Reply:
> I'm not sure I understand this comment, do you want me to add braces around the function body and add a return statement? If not, can you provide an example of what you mean?

Yep, thats what i meant. I'm personaly not an fan of such expressions.

Reply:
I guess you can put this into prettier(?) or eslint config, so you don't have to remind people ...

Reply:
✔️